### ISSUE TITLE
Rebuilding native modules directory naming issue

### ISSUE THREAD

**Issue by oransel** [electronjs/electron-rebuild#33]
When rebuilding with the serialport module, it names the final directory as 

```
electron-v0.34.1-darwin-x64
```

However, when ran with the executable, it is looking for

```
node-v46-darwin-x64
```

It seems like in main.js, the directory name and electron versioning is mixed.

**Comment by paulcbetts** [comment]
Nothing we can do about this, file on serialport?

**Comment by oransel** [comment]
how come it is about the serialport module? doesn't electron-rebuild create the output directory?

**Comment by paulcbetts** [comment]
Nope

**Comment by oransel** [comment]
As you can see, from npm it names the directory correctly. However, when ran with electron-build, the directory name changes.

![from-electron-rebuild](https://cloud.githubusercontent.com/assets/1380693/10776930/5eb66c9c-7cd3-11e5-8b19-0c394512941d.png)
![from-npm](https://cloud.githubusercontent.com/assets/1380693/10776931/5eb70cc4-7cd3-11e5-97ff-047ad514a28d.png)

**Comment by paulcbetts** [comment]
electron-rebuild just asks node-gyp to build your project. As I said again, serialport controls this

**Comment by BafS** [comment]
I have exactly the same problem with sqlite3.

For information: 
After having installed sqlite and electron-rebuild I did : `./node_modules/.bin/electron-rebuild` and it created a .node file at `node_modules/sqlite3/lib/binding/electron-v0.36-darwin-x64/node_sqlite3.node`. But electron give this error "*Uncaught Error: Cannot find module '/Users/me/dev/proj/node_modules/sqlite3/lib/binding/node-v47-darwin-x64/node_sqlite3.node*'"

I renamed the folder to `node-v47-darwin-x64` and it worked. 

Note that in `node_modules/sqlite3/build/config.gypi` I have the right path
```js
...
"module_path": "../lib/binding/node-v47-darwin-x64",
...
```

and `node_modules/sqlite3/package.json`

```js
...
  "binary": {
    "module_name": "node_sqlite3",
    "module_path": "./lib/binding/{node_abi}-{platform}-{arch}",
    "host": "https://mapbox-node-binary.s3.amazonaws.com",
    "remote_path": "./{name}/v{version}/{toolset}/",
    "package_name": "{node_abi}-{platform}-{arch}.tar.gz"
  },
...
```

PS: This command work (because module_path hard coded): `cd node_modules/sqlite3 && node-gyp rebuild --target=0.36.1 --arch=x64 --target_platform=darwin --runtime=electron --module_name=node_sqlite3 --module_path=../lib/binding/node-v47-darwin-x64`

**Comment by Malgalad** [comment]
Had the same issue, resolution: https://github.com/mapbox/node-pre-gyp/issues/189#issuecomment-173426428

**Comment by barbalex** [comment]
same problem here, only the resolution is none as the pull request is being ignored

**Comment by springmeyer** [comment]
This is now solved upstream by https://github.com/mapbox/node-pre-gyp/pull/187 - available in node-pre-gyp >= 0.6.25. So, the next step is to ensure that modules you are using that depend and bundle node-pre-gyp get re-tagged with the latest version.
